core/src/main/java/org/apache/iceberg/deletes/BitmapPositionDeleteIndex.java

```diff
@@ -27,6 +27,15 @@ class BitmapPositionDeleteIndex implements PositionDeleteIndex {
     roaring64Bitmap = new Roaring64Bitmap();
   }
 
+  void merge(PositionDeleteIndex other) {
```

Comment:
Why not just only allow BitmapPositionDeleteIndex here? Do we not have the type when we call merge?

Reply:
There is also `EmptyPositionDeleteIndex`. I actually started with `BitmapPositionDeleteIndex`. I may need to go back and check with fresh eyes.

Reply:
I switched to accepting `BitmapPositionDeleteIndex` here and moved casts to the utility class.